extensions/sql-migration/src/dialog/skuRecommendationResults/getAzureRecommendationDialog.ts

```diff
@@ -0,0 +1,402 @@
+		const container = _view.modelBuilder.flexContainer().withProps({
+			CSSStyles: {
+				'flex-direction': 'column',
+				'display': 'none',
+			}
+		}).component();
+
+		const instructions = _view.modelBuilder.text().withProps({
+			value: constants.AZURE_RECOMMENDATION_OPEN_EXISTING_FOLDER,
+			CSSStyles: {
+				...styles.LABEL_CSS,
+				'margin-bottom': '8px',
+			}
+		}).component();
+
+		const selectFolderContainer = _view.modelBuilder.flexContainer().withProps({
+			CSSStyles: {
+				'flex-direction': 'row',
+				'align-items': 'center',
+			}
+		}).component();
+
+		this._openExistingFolderInput = _view.modelBuilder.inputBox().withProps({
+			placeHolder: constants.FOLDER_NAME,
+			readOnly: true,
+			width: 320,
+			CSSStyles: {
+				'margin-right': '12px'
+			},
+		}).component();
+		this._disposables.push(this._openExistingFolderInput.onTextChanged(async (value) => {
+			if (value) {
+				this.migrationStateModel._skuRecommendationPerformanceLocation = value.trim();
+				this.dialog!.okButton.enabled = true;
+			}
+		}));
+
+		const openButton = _view.modelBuilder.button().withProps({
+			label: constants.OPEN,
+			width: 100,
+			CSSStyles: {
+				'margin': '0'
+			}
+		}).component();
+		this._disposables.push(openButton.onDidClick(async (e) => {
+			let folder = await this.handleBrowse();
+			this._openExistingFolderInput.value = folder;
+		}));
+
+		selectFolderContainer.addItems([
+			this._openExistingFolderInput,
+			openButton,
+		]);
+		container.addItems([
+			instructions,
+			selectFolderContainer,
+		]);
+		return container;
+	}
+
+	private async switchDataSourceContainerFields(containerType: PerformanceDataSourceOptions): Promise<void> {
+		this._performanceDataSource = containerType;
+
+		let okButtonEnabled = false;
+		switch (containerType) {
+			case PerformanceDataSourceOptions.CollectData: {
+				await this._collectDataContainer.updateCssStyles({ 'display': 'inline' });
+				await this._openExistingContainer.updateCssStyles({ 'display': 'none' });
+
+				if (this._collectDataFolderInput.value) {
+					okButtonEnabled = true;
+				}
+				break;
+			}
+			case PerformanceDataSourceOptions.OpenExisting: {
+				await this._collectDataContainer.updateCssStyles({ 'display': 'none' });
+				await this._openExistingContainer.updateCssStyles({ 'display': 'inline' });
+
+				if (this._openExistingFolderInput.value) {
+					okButtonEnabled = true;
+				}
+				break;
+			}
+		}
+		this.dialog!.okButton.enabled = okButtonEnabled;
+	}
+
+	public async openDialog(dialogName?: string) {
+		if (!this._isOpen) {
+			this._isOpen = true;
+			this.dialog = azdata.window.createModelViewDialog(constants.GET_AZURE_RECOMMENDATION, 'GetAzureRecommendationsDialog', 'narrow');
+
+			this.dialog.okButton.label = GetAzureRecommendationDialog.StartButtonText;
+			this._disposables.push(this.dialog.okButton.onClick(async () => await this.execute()));
+			this.dialog.cancelButton.onClick(() => this._isOpen = false);
```

Comment:
please wrap onClick() with
this._disposables.push(

Reply:
Addressing in another PR https://github.com/microsoft/azuredatastudio/pull/18345